### PR TITLE
Encapsulate code with proper OpenSSL version

### PR DESCRIPTION
**Pull request by mahantesh-nvidia**
Signed-off-by: Mahantesh Salimath <mahantesh@mellanox.com>

### CODE REVIEW

File: src/engine/e_bluefield.c [hunk 1]
```diff
@@ -45,24 +48,35 @@ static int engine_pka_init(ENGINE *e);
...
-/* BN operations */
 /* BN mod_exp */
 static int engine_pka_bn_mod_exp(BIGNUM *r, const BIGNUM *a, const BIGNUM *p,
                                  const BIGNUM *m, BN_CTX *ctx, BN_MONT_CTX *m_ctx);
-/* BN mod_inv */
-static int
-engine_pka_bn_mod_inv(const EC_GROUP *group, BIGNUM *r, const BIGNUM *x,
-                      BN_CTX *ctx);
 
-# if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
```
Comment (kblaiech) [inline src/engine/e_bluefield.c:57]: Any reason why we removed those guards?

File: src/engine/e_bluefield.c [hunk 2]
```diff
@@ -45,24 +48,35 @@ static int engine_pka_init(ENGINE *e);
...
+/* DSA stuff */
+static int engine_pka_dsa_init(DSA *dsa);
+static int engine_pka_dsa_finish(DSA *dsa);
+
 /* RSA stuff */
-#ifndef NO_RSA
```
Comment (kblaiech) [inline src/engine/e_bluefield.c:59]: same question?

File: src/engine/e_bluefield.c (hunk 1 above)
Comment (kblaiech) [reply src/engine/e_bluefield.c:57]: It is still valid, isn't? my only concern is when customers install an unsupported OpenSSL version, are they able to build the PKA software. I believe it is safe/harmless to leave such define, am I wrong?

Comment (kblaiech) [reply src/engine/e_bluefield.c:57]: Oops, never mind. I thought this define is for OpenSSL 1.0.0

File: src/engine/e_bluefield.c (hunk 2 above)
Comment (kblaiech) [reply src/engine/e_bluefield.c:59]: And what about this one? My understanding is OpenSSL has various build options. What if RSA config is excluded?

Comment (mahantesh-nvidia) [reply src/engine/e_bluefield.c:59]: Yes, right. I'll add this back.

Comment (mahantesh-nvidia) [reply src/engine/e_bluefield.c:59]: I have added the guards, Thanks.

**Review by kblaiech — APPROVED** [review]

**Pull request merged**